### PR TITLE
IEEE-143: created captcha template

### PR DESCRIPTION
**Pull request by arshinmar**
## Overview

- Resolves #IEEE-143
- A near carbon copy of https://github.com/ieeeuoft/newhacks-site-2021/pull/31
- Simply set a "" label for the Captcha field to fix wonky Captcha text appearing.

## Unit Tests Created

- Simply updated tests that required a filled-out application to include relevant captcha fields.

## Steps to QA

- Even though this is already checked, make sure new app form online looks okay.

### CODE REVIEW

File: hackathon_site/hackathon_site/settings/__init__.py [hunk 1]
```diff
@@ -50,6 +50,27 @@
...
+    SILENCED_SYSTEM_CHECKS = ["captcha.recaptcha_test_key_error"]
+else:
+    # If the default test keys are used, the captcha package will create a system
+    # check warning.
+    RECAPTCHA_PUBLIC_KEY = os.environ.get(
+        "RECAPTCHA_PUBLIC_KEY", RECAPTCHA_TEST_PUBLIC_KEY
+    )
+    RECAPTCHA_PRIVATE_KEY = os.environ.get(
+        "RECAPTCHA_PRIVATE_KEY", RECAPTCHA_TEST_PRIVATE_KEY
+    )
```
Comment (grahamhoyes) [inline hackathon_site/hackathon_site/settings/__init__.py:72]:
```suggestion
    RECAPTCHA_PUBLIC_KEY = os.environ["RECAPTCHA_PUBLIC_KEY"]
    RECAPTCHA_PRIVATE_KEY = os.environ["RECAPTCHA_PRIVATE_KEY"]
```

File: hackathon_site/hackathon_site/settings/__init__.py [hunk 2]
```diff
@@ -50,6 +50,23 @@
 
 CORS_ALLOW_CREDENTIALS = True
 
+RECAPTCHA_DOMAIN = "www.recaptcha.net"
+
+RECAPTCHA_TEST_PUBLIC_KEY = "6LeIxAcTAAAAAJcZVRqyHh71UMIEGNQ_MXjiZKhI"
+RECAPTCHA_TEST_PRIVATE_KEY = "6LeIxAcTAAAAAGG-vFI1TnRWxMZNFuojJ4WifJWe"
+
+if DEBUG:
```
Comment (grahamhoyes) [inline hackathon_site/hackathon_site/settings/__init__.py]: My bad, I forgot debug is false during testing. Change this to `if DEBUG or IN_TESTING` and all should be well

File: README.md [hunk 3]
```diff
@@ -67,6 +67,7 @@ In order to run the django and react development servers locally (or run tests),
 | DB_NAME        |                                   | hackathon_site    | Postgres database name.                                                           |
 | REDIS_URI      |                                   | 172.17.0.1:6379/1 | Redis [URI](https://github.com/lettuce-io/lettuce-core/wiki/Redis-URI-and-connection-details#uri-syntax). `<host>:<port>/<database>`. |
 | **REACT_APP_DEV_SERVER_URL** | http://localhost:8000 |                 | Path to the django development server, used by React. Update the port if you aren't using the default 8000. |
+| RECAPTCHA_PUBLIC_KEY | Something | In development, a key that will skip the challenge | Key info: https://www.google.com/recaptcha/ |
```
Comment (grahamhoyes) [inline README.md]: `RECAPTCHA_PRIVATE_KEY` is also required. I guess we can put "A reCAPTCHA public key" / "a reCAPTCHA private key" in the required value column

File: hackathon_site/hackathon_site/settings/__init__.py [hunk 4]
```diff
@@ -50,6 +50,23 @@
 
 CORS_ALLOW_CREDENTIALS = True
 
+RECAPTCHA_DOMAIN = "www.recaptcha.net"
+
+RECAPTCHA_TEST_PUBLIC_KEY = "6LeIxAcTAAAAAJcZVRqyHh71UMIEGNQ_MXjiZKhI"
+RECAPTCHA_TEST_PRIVATE_KEY = "6LeIxAcTAAAAAGG-vFI1TnRWxMZNFuojJ4WifJWe"
+
+if DEBUG or IN_TESTING:
```
Comment (grahamhoyes) [inline hackathon_site/hackathon_site/settings/__init__.py:58]: Ok my bad, I forgot that `IN_TESTING` doesn't get changed to true until after this file finishes running, should probably have made that an environment variable originally.

You can change the keys below back to `os.environ.get("RECAPTCHA_PUBLIC_KEY", RECAPTCHA_TEST_PUBLIC_KEY)` etc. I tested, and the systems check framework will actually prevent the site from running, so nobody is at risk of accidentally using the test keys in production.

**Review by Leo6Leo — APPROVED** [review]
1. Tested locally, works without any issues.
2. All tests passed
3. Read through all changed documents, did not find any issues!!! 
Perfecto!

**Comment by arshinmar** [comment]
@Leo6Leo thanks!

**Pull request merged**